### PR TITLE
Open NodePort range in Azure network security group

### LINKED ISSUE #7962: Open NodePorts for Azure Network Security Group

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**: Opens up the configured node port range in the Azure NSG created by seed-controller-manager.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7962

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Open Azure Network Security Group for external traffic to NodePort port ranges
```

Signed-off-by: Marvin Beckers <marvin@kubermatic.com>

### CODE REVIEW

**Review by moadqassem — COMMENTED** [review]
/test pre-kubermatic-e2e-azure-ubuntu-1.22

File: pkg/provider/cloud/azure/provider.go [hunk 1]
```diff
@@ -415,6 +415,20 @@ func (a *Azure) ensureSecurityGroup(cloud kubermaticv1.CloudSpec, location strin
 						Priority:                 to.Int32Ptr(300),
 					},
 				},
+				{
+					// Allow access to node ports from everywhere
+					Name: to.StringPtr("node_ports_ingress"),
+					SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+						Direction:                network.SecurityRuleDirectionInbound,
+						Protocol:                 network.SecurityRuleProtocolTCP,
```
Comment (kron4eg) [inline pkg/provider/cloud/azure/provider.go]: why discriminate against UDP nodeports?

Comment (embik) [reply pkg/provider/cloud/azure/provider.go]: I looked at the OpenStack rule (https://github.com/kubermatic/kubermatic/blob/master/pkg/provider/cloud/openstack/helper.go#L214) and it looked like it enables TCP only too. Happy to add a rule for UDP if you want.

Comment (kron4eg) [reply pkg/provider/cloud/azure/provider.go]: I meant it could be: `Protocol: "*"`

Comment (embik) [reply pkg/provider/cloud/azure/provider.go]: Good point! Updated to `network.SecurityRuleProtocolAsterisk`.

**Review by kron4eg — APPROVED** [review]
/approve
/lgtm

**Pull request merged**